pages/docs/manual/latest/generate-converters-accessors.mdx

```diff
@@ -0,0 +1,565 @@
+
+```resi
+let coordinatesToJs: coordinates => abs_coordinates;
+
+let coordinatesFromJs: abs_coordinates => coordinates;
+```
+
+#### Usage
+
+Using `newType`, you've now prevented consumers from inadvertently doing the following:
+
+```res
+let myCoordinates = {
+  x: 10,
+  y: 20
+};
+let jsCoords = coordinatesToJs(myCoordinates);
+
+let x = jsCoords["x"]; /* disallowed! Don't access the object's internal details */
+```
+
+Same generated output. Isn't it great that types prevent invalid accesses you'd otherwise have to encode at runtime?
+
+## Generate Converters for JS Integer Enums and Variants
+
+Use `@bs.deriving(jsConverter)` on a variant type to create converter functions that allow back and forth conversion between JS integer enum and ReScript variant values.
+
+```res
+@bs.deriving(jsConverter)
+type fruit =
+  | Apple
+  | Orange
+  | Kiwi
+  | Watermelon;
+```
+
+This option causes `jsConverter` to, again, generate functions of the following types:
+
+```resi
+let fruitToJs: fruit => int;
+
+let fruitFromJs: int => option(fruit);
+```
+
+For `fruitToJs`, each fruit variant constructor would map into an integer, starting at 0, in the order they're declared.
+
+For `fruitFromJs`, the return value is an `option`, because not every int maps to a constructor.
+
+You can also attach a `@bs.as(1234)` to each constructor to customize their output.
+
+### Usage
+
+```res
+@bs.deriving(jsConverter)
+type fruit =
+  | Apple
+  | @bs.as(10) Orange
+  | @bs.as(100) Kiwi
+  | Watermelon
+
+let zero = fruitToJs(Apple) /* 0 */
+
+switch fruitFromJs(100) {
+| Some(Kiwi) => Js.log("this is Kiwi")
+| _ => Js.log("received something wrong from the JS side")
+}
+```
+
+**Note**: by using `@bs.as` here, all subsequent number encoding changes. `Apple` is still `0`, `Orange` is `10`, `Kiwi` is `100` and `Watermelon` is **`101`**!
+
+### More Safety
+
+Similar to the JS object <-> record deriving, you can hide the fact that the JS enum are ints by using the same `newType` option with `@bs.deriving(jsConverter)`:
+
+```res
+@bs.deriving({jsConverter: newType})
+type fruit =
+  | Apple
+  | @bs.as(100) Kiwi
+  | Watermelon;
+```
+
+This option causes `@bs.deriving(jsConverter)` to generate functions of the following types:
+
+```resi
+let fruitToJs: fruit => abs_fruit;
+
+let fruitFromJs: abs_fruit => fruit;
+```
+
+For `fruitFromJs`, the return value, unlike the previous non-abstract type case, doesn't contain an `option`, because there's no way a bad value can be passed into it; the only creator of `abs_fruit` values is `fruitToJs`!
+
+#### Usage
+
+```res
+@bs.deriving({jsConverter: newType})
+type fruit =
+  | Apple
+  | @bs.as(100) Kiwi
+  | Watermelon
+
+let opaqueValue = fruitToJs(Apple)
+
+@bs.module("myJSFruits") external jsKiwi: abs_fruit = "iSwearThisIsAKiwi"
+let kiwi = fruitFromJs(jsKiwi)
+
+let error = fruitFromJs(100) /* nope, can't take a random int */
+```
+
+## Generate Converters for JS String Enums and Polymorphic Variants
+
```

Comment:
The best practice: use jsConverter to convert between int and rescript types for regular variants. For string converter, it is not needed any more in ReScript

Reply:
@bobzhang I believe we are still missing polyvariant docs to properly reflect that best practise, but I will add a version notice similar to the JS object converter

Reply:
added a note that since ReScript  >= 8.2 compiles polyvars to strings